File: app/src/androidTest/java/ru/roscha_akademii/medialib/common/MockMediaLibApplication.java
package ru.roscha_akademii.medialib.common;

import android.util.Log;

public class MockMediaLibApplication extends MediaLibApplication {
    public void setTestComponent(ApplicationComponent component) {
        this.setTestComponent(component);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d("happy", "MockMediaLibApplication is running");
    }
}

File: app/src/androidTest/java/ru/roscha_akademii/medialib/common/MockMediaLibApplication.kt
package ru.roscha_akademii.medialib.common

import android.util.Log

class MockMediaLibApplication : MediaLibApplication() {

    fun setTestComponent(testComponent: ApplicationComponent)
    {
        _component = testComponent
    }

    override fun onCreate() {
        super.onCreate()
        Log.d("happy", "MockMediaLibApplication is running")
    }
}